www/include/reporting/dashboard/common-Func.php

```diff
@@ -38,29 +38,58 @@
  */
 function getPeriodToReport()
 {
-    $period = (isset($_POST["period"])) ? $_POST["period"] : "";
-    $period = (isset($_GET["period"])) ? $_GET["period"] : $period;
-    $period_choice = (isset($_POST["period_choice"])) ? $_POST["period_choice"] : "";
-    $end_date = 0;
-    $start_date = 0;
-    $start_date = (isset($_POST["StartDate"])) ? $_POST["StartDate"] : "";
-    $start_date = (isset($_GET["start"])) ? $_GET["start"] : $start_date;
-    $end_date = (isset($_POST["EndDate"])) ? $_POST["EndDate"] : "";
-    $end_date = (isset($_GET["end"])) ? $_GET["end"] : $end_date;
+    $period = '';
+    $start_date = '';
+    $end_date = '';
+    $period_choice = '';
+
+    if (isset($_POST['period'])) {
+        $period = filter_var($_POST['period'], FILTER_SANITIZE_STRING);
+    } elseif (isset($_GET['period'])) {
+        $period = filter_var($_GET['period'], FILTER_SANITIZE_STRING);
+    }
+
+    if (isset($_POST['period_choice'])) {
+        $period_choice = filter_var($_POST['period_choice'], FILTER_SANITIZE_STRING);
+    }
+
+    if (isset($_POST['StartDate'])) {
+        $start_date = filter_var($_POST['StartDate'], FILTER_SANITIZE_STRING);
+    } elseif (isset($_GET['StartDate'])) {
+        $start_date = filter_var($_GET['StartDate'], FILTER_SANITIZE_STRING);
+    }
+
+    if (isset($_POST['EndDate'])) {
+        $end_date = filter_var($_POST['EndDate'], FILTER_SANITIZE_STRING);
+    } elseif (isset($_GET['EndDate'])) {
+        $end_date = filter_var($_GET['EndDate'], FILTER_SANITIZE_STRING);
+    }
+
     $interval = array(0, 0);
-    if ($period_choice == "custom" && $start_date != "" && $end_date != "") {
-        $period = "";
+
+    if (
+        $period_choice == 'custom' &&
+        $start_date != '' &&
+        $end_date != ''
+    ) {
+        $period = '';
     }
-    if ($period == "" && $start_date == "" && $end_date == "") {
-        $period = "yesterday";
+
+    if ($period == '' &&
+        $start_date == '' &&
+        $end_date == ''
+    ) {
+        $period = 'yesterday';
     }
-    if ($period == "" && $start_date != "") {
+
+    if ($period == '' && $start_date != '') {
         $interval = getDateSelectCustomized($start_date, $end_date);
     } else {
         $interval = getDateSelectPredefined($period);
     }
-    $start_date = $interval[0];
-    $end_date = $interval[1];
+
+    list($start_date, $end_date) = $interval;
+
     return(array($start_date, $end_date));
```

Comment:
why not return $interval ?
if interval contains only $start_date and $end_date

Reply:
Did not really thought it through :+1: 
I'll keep it this way for now